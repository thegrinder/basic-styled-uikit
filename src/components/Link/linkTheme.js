import { darken, lighten } from 'polished';
import colors from '../../theme/colors';

const linkTheme = {
  linkTypes: {
    default: {
      light: {
        normal: {
          color: colors.primary600,
        },
        hover: {
          color: darken(0.1, colors.primary600),
        },
      },
      dark: {
        normal: {
          color: colors.primary400,
        },
        hover: {
          color: lighten(0.1, colors.primary400),
        },
      },
    },
    muted: {
      light: {
        normal: {
          color: colors.neutral600,
        },
        hover: {
          color: darken(0.1, colors.neutral600),
        },
      },
      dark: {
        normal: {
          color: colors.neutral400,
        },
        hover: {
          color: lighten(0.1, colors.neutral400),
        },
      },
    },
  },
  common: {
    fontFamily: 'Lato, Arial, Helvetica, sans-serif',
  },
};

export default linkTheme;
