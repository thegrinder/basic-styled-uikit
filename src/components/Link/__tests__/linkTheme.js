import linkTheme from '../linkTheme';

describe('linkTheme', () => {
  it('should be defined', () => {
    expect(linkTheme).toBeDefined();
    expect(linkTheme).toMatchSnapshot();
  });
});
