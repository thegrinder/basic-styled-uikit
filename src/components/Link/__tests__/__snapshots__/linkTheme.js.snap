// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`linkTheme should be defined 1`] = `
Object {
  "common": Object {
    "fontFamily": "Lato, Arial, Helvetica, sans-serif",
  },
  "linkTypes": Object {
    "default": Object {
      "dark": Object {
        "hover": Object {
          "color": "#72bcf8",
        },
        "normal": Object {
          "color": "#42A5F5",
        },
      },
      "light": Object {
        "hover": Object {
          "color": "#166dba",
        },
        "normal": Object {
          "color": "#1E88E5",
        },
      },
    },
    "muted": Object {
      "dark": Object {
        "hover": Object {
          "color": "#ebeef0",
        },
        "normal": Object {
          "color": "#ced4da",
        },
      },
      "light": Object {
        "hover": Object {
          "color": "#6c757d",
        },
        "normal": Object {
          "color": "#868e96",
        },
      },
    },
  },
}
`;
